### ISSUE TITLE
Moodle PROXY setting not used

### ISSUE THREAD

**Issue by nadavkav** [davidpesce/moodle-logstore_xapi#84]
It seems that the RusticiSoftware/TinCanPHP library that is used in this project did not implement proxy support when initiating the remotelrs class (and obviously, when sending the xAPI staements to the remote LRS)

An issue report and a workaround can be found here:
https://github.com/RusticiSoftware/TinCanPHP/issues/63

I have temporarily worked around it, as our campus is behind a reverse proxy (firewall).
But a fix need to be applied first in RusticiSoftware/TinCanPHP before we can implement a permanent solution on our end. (this project)

Please help raise priority of this issue on the RusticiSoftware/TinCanPHP project (above link) as we are depending on them.

**Comment by ryasmi** [comment]
Thanks @nadavkav 👍

**Comment by deedey** [comment]
You are right @nadavkav.
Thank you for this suggest.
I think it would be much more appropriate when calling from "**RemoteLRS.php**" in "**/ mod / tincanlaunch / TinCanPHP / src /**", the API directly retrieves the proxy data if exists from the table "**mdl_config**" variables "**proxyhost**" and "**proxyport**" to implement them directly.
If this is managed at "RemoteLRS.php", the problem does not arise, nor for Tincanlaunch nor logstore_xapi.

**Comment by nadavkav** [comment]
@brianjmiller (of RusticiSofware) was quick enough to come up with a fix on their side.
https://github.com/RusticiSoftware/TinCanPHP/pull/68
I added a quick test code snippet inside the above issue, but suggest a better one on our side:
https://github.com/LearningLocker/xAPI-Recipe-Emitter/blob/master/src/Repository.php#L15

```
public function __construct(TinCanRemoteLrs $store) {
        global $CFG;
        $this->store = $store;
        if (!empty($CFG->proxyhost)) {
            $this->store->setProxy($CFG->proxyhost.':'.$CFG->proxyport);    
        }
    }
```

**Comment by deedey** [comment]
@garemoko  @brianjmiller @nadavkav 
Can you suggest a same solution to integrate proxy parameters automaticly in [https://github.com/garemoko/moodle-mod_tincanlaunch]
I can open an issue if necessary.

**Comment by ryasmi** [comment]
Hi @nadavkav, thanks for that snippet, but we cannot allow Moodle specific code inside the xAPI recipe emitter, because it is used by other plugins outside of Moodle. Therefore, I would recommend the following adjustment to the ["classes/log/store.php" file](https://github.com/xAPI-vle/moodle-logstore_xapi/blob/master/classes/log/store.php#L152).

``` php
    /**
     * Creates a connection the xAPI store.
     * @return xapi_repository
     */
    private function connect_xapi_repository() {
        global $CFG;
        $remote_lrs = new tincan_remote_lrs(
            $this->get_config('endpoint', ''),
            '1.0.1',
            $this->get_config('username', ''),
            $this->get_config('password', '')
        );
        if (!empty($CFG->proxyhost)) {
          $remote_lrs->setProxy($CFG->proxyhost.':'.$CFG->proxyport);
        }
        return new xapi_repository($remote_lrs);
    }
```

I'd recommend trying out that change, then submitting a pull request to this repo.

**Comment by deedey** [comment]
@ryansmith94 @nadavkav ,
I've tested this last part of  store.php script.
It works fine. Thanks a lot.

**Comment by nadavkav** [comment]
@ryansmith94 you are right. I thought It was a Moodle specific plugin.
Anyways, your suggest code looks great. 
And I already see @deedey is going to use it too.
So thank you all for quickly fixing this!

**Comment by ryasmi** [comment]
@deedey feel free to make a PR so that other people have it too 👍

**Comment by davidpesce** [comment]
This is fixed in v1.2.0
